python/pyspark/sql/dataframe.py

```diff
@@ -1980,6 +1980,9 @@ def dropDuplicates(self, subset=None):
         |Alice|  5|    80|
         +-----+---+------+
         """
+        if subset is not None and not isinstance(subset, (list, tuple)):
```

Comment:
Can we check if it's `Iterable` instead for now? Just in case it breaks other user's apps.

Reply:
What if we check if it is Iterable and not string?

Trying to avoid the case where a user just provides just a string and results in a spark error from providing dropDuplicates("col_name")

Reply:
Okay, seems like it only works with list and tuple by py4j type conversion. Seems fine

Reply:
Are you sure? I just tested that I was able to pass a generator into the dropDuplicates and that worked

Reply:
the thing is we don't support general iterators in other places, eg dropna

Reply:
yeah, the change seems fine.